src/components/LoadingScreen/LoadingScreen.js

```diff
@@ -1,22 +1,76 @@
 // @flow
-import React, { PureComponent } from 'react';
+import React, { Fragment, PureComponent } from 'react';
 import { connect } from 'react-redux';
 import styled from 'styled-components';
+import IconBase from 'react-icons-kit';
+import { info } from 'react-icons-kit/feather/info';
 
-import { getBlockingStatus } from '../../reducers/app-status.reducer';
+import Spacer from '../Spacer';
+import ProgressBar from '../ProgressBar';
+import {
+  getBlockingStatus,
+  getStatusText,
+} from '../../reducers/app-status.reducer';
 
 import guppyLoaderSrc from '../../assets/images/guppy-loader.gif';
 import { COLORS, Z_INDICES } from '../../constants';
 
+const INFOTEXT_HEIGHT = 80;
+
 type Props = {
   showLoadingScreen: boolean,
+  statusText: string,
+};
+
+type State = {
+  showStatus: boolean,
 };
 
-class LoadingScreen extends PureComponent<Props> {
+class LoadingScreen extends PureComponent<Props, State> {
+  state = {
+    showStatus: false,
+  };
+
+  progress: number = 0;
+
+  toggleStatus = () => {
+    this.setState(state => ({
+      showStatus: !state.showStatus,
+    }));
+  };
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.statusText !== this.props.statusText) {
+      // new status
+      const [_, total] = (/\d+\/\d+/.exec(nextProps.statusText) || [
+        '',
+      ])[0].split('/');
+      if (total) {
+        this.progress += 1 / parseInt(total);
+        // limit progress to 1
+        this.progress = Math.min(1, this.progress);
+      }
+    }
+    if (!this.props.showLoadingScreen) {
+      this.progress = 0;
+    }
+  }
+
   render() {
-    const { showLoadingScreen } = this.props;
+    const { showLoadingScreen, statusText } = this.props;
+    const { showStatus } = this.state;
+    const ellipsis = statusText.length > 220 ? '...' : '';
```

Comment:
I'm getting undefined for `statusText` so the loading screen never moves beyond the first message in the speech bubble. However it looks like the reinstall is working because I deleted my node_modules to test, and they successfully re-appear. 

<img width="400" alt="screen shot 2018-11-02 at 7 36 23 am" src="https://user-images.githubusercontent.com/17421347/47921567-2567d580-de72-11e8-8082-d49e85a92a2e.png">

<img width="400" alt="screen shot 2018-11-02 at 7 28 00 am" src="https://user-images.githubusercontent.com/17421347/47921056-e7b67d00-de70-11e8-99c0-080e04950f49.png">

It successfully exits:

<img width="460" alt="screen shot 2018-11-02 at 7 38 49 am" src="https://user-images.githubusercontent.com/17421347/47921659-61029f80-de72-11e8-9dac-6285887a4220.png">

Reply:
OK, I don't get that here. 
But I'll add a conditional or pass a default `statusText` prop so it won't fail.

Reply:
Please try again. It should work now.